File: heapsort/go/heapsort.go
package heapsort

func Sort(a []int, f MaxHeapifyFunc) {
	// fmt.Printf("Sort: %v\n", a)

	h := f(a)

	for i := len(a) - 1; i >= 0; i-- {
		a[i] = h.RemoveRoot()
	}
}

File: heapsort/javascript/heapsort.js
export default function sort(a, maxHeapify) {
    if (!a) {
        return;
    }

    const h = maxHeapify(a);

    for (let i = a.length - 1; i >= 0; i -= 1) {
        a[i] = h.deleteMax();
    }
}
